### PR TITLE
Fix flake8 issues on v22, v23 and v24rst

### LINKED ISSUE #24200: DOC: Fix flake8 issues in doc/source/whatsnew/v0.24.0.rst
We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

The first step of this issue would be edit `setup.cfg` in the pandas home, and in the flake8-rst section, remove from the `exclude` list the file `doc/source/whatsnew/v0.24.0.rst`

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version 0.7.0 or higher):

```
$ flake8-rst doc/source/whatsnew/v0.24.0.rst 
source/whatsnew/v0.24.0.rst:281:26: E127 continuation line over-indented for visual indent
source/whatsnew/v0.24.0.rst:294:4: E999 IndentationError: unexpected indent
source/whatsnew/v0.24.0.rst:294:5: E303 too many blank lines (2)
source/whatsnew/v0.24.0.rst:294:5: E111 indentation is not a multiple of four
source/whatsnew/v0.24.0.rst:294:5: E113 unexpected indentation
source/whatsnew/v0.24.0.rst:444:92: E501 line too long (96 > 79 characters)
source/whatsnew/v0.24.0.rst:452:92: E501 line too long (80 > 79 characters)
source/whatsnew/v0.24.0.rst:561:84: E501 line too long (80 > 79 characters)
source/whatsnew/v0.24.0.rst:571:84: E501 line too long (84 > 79 characters)
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from setup.cfg. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.
cc: @datapythonista

### LINKED ISSUE #24201: DOC: Fix flake8 issues in doc/source/whatsnew/v0.23.0.rst
We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

The first step of this issue would be edit `setup.cfg` in the pandas home, and in the flake8-rst section, remove from the `exclude` list the file `doc/source/whatsnew/v0.23.0.rst`

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version 0.7.0 or higher):

```
$ flake8-rst doc/source/whatsnew/v0.23.0.rst 
source/whatsnew/v0.23.0.rst:72:18: E999 SyntaxError: invalid syntax
source/whatsnew/v0.23.0.rst:339:83: E501 line too long (80 > 79 characters)
source/whatsnew/v0.23.0.rst:699:84: E501 line too long (87 > 79 characters)
source/whatsnew/v0.23.0.rst:828:84: E501 line too long (94 > 79 characters)
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from setup.cfg. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.
cc: @datapythonista

### PR DESCRIPTION
**Pull request by thoo**
- [x] closes #24200, #24201 and #24202 
- [x] `flake8-rst source/whatsnew/v0.22.0.rst` and `flake8-rst source/whatsnew/v0.23.0.rst` 
This is related to #24177

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.rst [hunk 1]
```diff
@@ -336,7 +336,8 @@ outside the existing valid values while preserving those inside.  (:issue:`16284
 
 .. ipython:: python
 
-   ser = pd.Series([np.nan, np.nan, 5, np.nan, np.nan, np.nan, 13, np.nan, np.nan])
+   ser = pd.Series([np.nan, np.nan, 5, np.nan, np.nan,
+                   np.nan, 13, np.nan, np.nan])
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.23.0.rst]: I think this is missing one more space

**Review by datapythonista — COMMENTED** [review]
looks good, thanks for the fixes @thoo 

just one wrong indentation I think

File: doc/source/whatsnew/v0.23.0.rst (hunk 1 above)
Comment (thoo) [reply doc/source/whatsnew/v0.23.0.rst]: Thanks. Done.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @thoo

**Review by datapythonista — APPROVED** [review]
looks good, thanks @thoo 

but in general, once you've got a PR approved, better to open a new one, instead of modifying it

**Comment by thoo** [comment]
@datapythonista Thanks. I will definitely do it next time.

**Pull request merged**